bug fix

ivywei0125 opened the pull request:
What does this implement/fix? Explain your changes.
---------------------------------------------------
1. fix wrong operation stack bug [AB#1974846](https://dev.azure.com/mseng/VSJava/_sprints/backlog/Java%20Azure%20Tooling/VSJava/CY2022/2022-09?workitem=1974846)
2. update select status of subscriptions  [AB#1983740](https://dev.azure.com/mseng/VSJava/_sprints/backlog/Java%20Azure%20Tooling/VSJava/CY2022/2022-09?workitem=1983740)
3. update message [AB#1948968](https://dev.azure.com/mseng/VSJava/_sprints/backlog/Java%20Azure%20Tooling/VSJava/CY2022/2022-09?text=insigh&workitem=1948968)
4. [AB#1976033](https://dev.azure.com/mseng/VSJava/_sprints/backlog/Java%20Azure%20Tooling/VSJava/CY2022/2022-09?workitem=1976033)

Does this close any currently open issues?
------------------------------------------


Any relevant logs, screenshots, error output, etc.?
-------------------------------------


Any other comments?
-------------------


Has this been tested?
---------------------------
- [ ] Tested

wangmingliang-ms commented on azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/operation/OperationThreadContext.java:
```diff
@@ -55,6 +55,10 @@ public Operation currentOperation() {
         return this.operation;
     }
 
+    public void resetThreadId() {
```
don't make it `public`

wangmingliang-ms commented on azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/Account.java:
```diff
@@ -190,8 +190,8 @@ public void setSelectedSubscriptions(List<String> selectedSubscriptionIds) {
-        this.getSubscriptions().forEach(s -> s.setSelected(false));
-        this.getSubscriptions().stream()
+        this.subscriptions.forEach(s -> s.setSelected(false));
+        this.subscriptions.stream()
```
does this change have any effects?

wangmingliang-ms commented on azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/operation/OperationThreadContext.java:
```diff
@@ -114,13 +114,19 @@ private synchronized void dispose() {
         final long threadId = Thread.currentThread().getId();
         assert this == current && this.threadId == threadId : String.format("[threadId:%s] disposing context[%s] in context[%s].", threadId, this, current);
-        if (this.parent == null || this.threadId != this.parent.threadId) { // this is the root task of current thread.
+        if (this.parent == null || this.threadId != this.parent.threadId || this.parent.threadId == -1) { // this is the root task of current thread.
```
should also check ancestors' threadId.

ivywei0125 committed:
#1976033 doValidate again when change is false and status is validating

ivywei0125 committed:
#1948968 update message

ivywei0125 committed:
block() function is not allowed to run in nio thread.
use subscribe() to get string in non-blocking way.

ivywei0125 committed:
#1974846 when remove current thread context, we should reset thread id
if there are children context of removed context whose thread id is -1, it should be removed too

ivywei0125 committed:
#1974846 simplify

ivywei0125 committed:
#1942819 check if region available when create

ivywei0125 committed:
#1974846 save to var to decrease redundant call

Flanker32 reviewed (approved):
LGTM
